sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1576,7 +1576,8 @@ object SQLConf {
       .doc("Prune nested fields from a logical relation's output which are unnecessary in " +
         "satisfying a query. This optimization allows columnar file format readers to avoid " +
         "reading unnecessary nested column data. Currently Parquet and ORC are the " +
-        "data sources that implement this optimization.")
+        "data sources that implement this optimization. This optimization also allows pruning " +
+        "unnecessary nested fields in expressions of operator.")
```

Comment:
Can we have another configuration for this `pruning on expressions`? For me, this is not about `nestedSchemaPruning` because this seems irrelevant `schema` changing.

Reply:
We talked about it in previous discussion, that is an option. Reusing `nestedSchemaPruning` was to simplify configs. Let me create a separate config for this.

Reply:
Sorry about asking this. During last review, I realized that this can disable all the other nested schema pruning features to the users.